### DISCUSSION TITLE
Multi stage forms

### DISCUSSION THREAD

**Discussion by jacobluke89** [react-hook-form/react-hook-form#3869] [Q&A] [answered]
Hi there, 

I have a form, that is made up of individual stages i.e.

draft -> pending approval -> pending order -> pending close -> closed

The form is populated by api's that I've created. We have fields that are then populated for the next stage, from the previous stage. These fields include inputs, textAreas and dropdowns. I have managed to get all the fields validating on draft submission. 

However, in pend approval, it suggests that the fields are not valid as the watchers do not have any data. is there a way to set the watchers with the data from the api, which will mean it won't fail on submission to the next step? 

Thank you for your help. :)

**Comment by bluebill1049** [comment]
can you share a simple codesandbox?

**Reply by jacobluke89** [reply]
Hi bluebill,
Thank you for your response. I worked it out. 
Basically in The Controller component, I wasn't telling it what the default value was, which meant it thought it was empty! 

Thank you for the recommendation of the sandbox, whilst writing it, i realised that was where I was going wrong! 

https://codesandbox.io/s/stoic-shaw-88300?file=/src/App.js

**Comment by jacobluke89** [comment, marked as answer]
I worked out the answer myself, I wasn't passing the defaultValue through. 

https://codesandbox.io/s/stoic-shaw-88300?file=/src/App.js

**Reply by bluebill1049** [reply]
sweet! awesome work.
